core/models/tool_configuration.rb

```diff
@@ -30,6 +30,8 @@ def save
       file.write(YAML.dump(@config))
       file.close
       config_file = File.expand_path(CONFIG_FILE_NAME, XDG['CONFIG_HOME'].to_s)
+      FileUtils.mkpath(File.dirname(config_file)) unless Dir.exist?(File.dirname(config_file))
```

Comment:
1. You introduce the duplication, `File.dirname(config_file)`, in a single file.
2. You must check the return value of the `mkpath` (rename it to `mkdir_p`). This method will in many cases will work correctly, but in some cases it must notice user that something went wrong.